debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/transforms/MongoDataConverter.java

```diff
@@ -0,0 +1,327 @@
+                        double temp = arrValue.asDouble().getValue();
+                        list.add(temp);
+                    } else if (arrValue.getBsonType() == BsonType.BINARY && valueType == BsonType.BINARY) {
+                        byte[] temp = arrValue.asBinary().getData();
+                        list.add(temp);
+                    } else if (arrValue.getBsonType() == BsonType.INT32 && valueType == BsonType.INT32) {
+                        int temp = arrValue.asInt32().getValue();
+                        list.add(temp);
+                    } else if (arrValue.getBsonType() == BsonType.INT64 && valueType == BsonType.INT64) {
+                        long temp = arrValue.asInt64().getValue();
+                        list.add(temp);
+                    } else if (arrValue.getBsonType() == BsonType.DATE_TIME && valueType == BsonType.DATE_TIME) {
+                        long temp = arrValue.asInt64().getValue();
+                        list.add(temp);
+                    } else if (arrValue.getBsonType() == BsonType.DECIMAL128 && valueType == BsonType.DECIMAL128) {
+                        String temp = arrValue.asDecimal128().getValue().toString();
+                        list.add(temp);
+                    } else if (arrValue.getBsonType() == BsonType.TIMESTAMP && valueType == BsonType.TIMESTAMP) {
+                        int temp = arrValue.asInt32().getValue();
+                        list.add(temp);
+                    } else if (arrValue.getBsonType() == BsonType.BOOLEAN && valueType == BsonType.BOOLEAN) {
+                        boolean temp = arrValue.asBoolean().getValue();
+                        list.add(temp);
+                    } else if (arrValue.getBsonType() == BsonType.DOCUMENT && valueType == BsonType.DOCUMENT) {
+                        Schema schema1 = schema.field(keyvalueforStruct.getKey()).schema().valueSchema();
+                        Struct struct1 = new Struct(schema1);
+                        for (Entry<String, BsonValue> entry9 : arrValue.asDocument().entrySet()) {
+                            convertFieldValue(entry9, struct1, schema1);
+                        }
+                        list.add(struct1);
+                    }
+                });
+                colValue = list;
+            }
+            break;
+
+        default:
+            break;
+        }
+        struct.put(key, keyvalueforStruct.getValue().isNull() ? null : colValue);
+    }
+
+    public static void addFieldSchema(Entry<String, BsonValue> keyValuesforSchema, SchemaBuilder builder) {
+
+        String key = keyValuesforSchema.getKey();
+        BsonType type = keyValuesforSchema.getValue().getBsonType();
+
+        switch (type) {
+
+        case NULL:
+            log.warn("Data type {} not currently supported", type);
+            break;
+
+        case STRING:
+        case JAVASCRIPT:
+        case OBJECT_ID:
+        case DECIMAL128:
+            builder.field(key, Schema.STRING_SCHEMA);
```

Comment:
Having played a bit more with this feature, I've come to think that we should call `optional()` for all the field schemas. Otherwise, the corresponding columns will be created as `NOT NULL` by the JDBC sink connector, which is not always what we want.

Reply:
i will be changing all the field schemas to optional.